### ISSUE TITLE
CKV_AWS_111 & CKV_AWS_109 Only work when Effect is Deny

### ISSUE THREAD

**Issue by hpapadopoulos** [bridgecrewio/checkov#1392]
**Describe the bug**
Looks like checks:
- Check: CKV_AWS_111: "Ensure IAM policies does not allow write access without constraints"
- Check: CKV_AWS_109: "Ensure IAM policies does not allow permissions management / resource exposure without constraints"

Fail although there are conditions in place when the effect is `Allow`

**To Reproduce**
Failing although should pass:
```
data "aws_iam_policy_document" "wham-dev" {
  statement {
    sid       = "SsoInLinePolicy1"
    effect    = "Allow"
    actions   = [
      "quicksight:*",
      "ds:AuthorizeApplication",
      "ds:UnauthorizeApplication",
      "ds:CheckAlias",
      "ds:CreateAlias",
      "ds:DescribeDirectories",
      "ds:DescribeTrusts",
      "ds:DeleteDirectory",
      "ds:CreateIdentityPoolDirectory",
      "iam:ListAccountAliases"
    ]
    resources = ["*"]
    condition {
      test     = "StringEquals"
      variable = "aws:RequestTag/Tribe"
      values = [
        "someTribe",
      ]
    }
  }
  statement {
    sid       = "SsoInLinePolicy2"
    effect    = "Allow"
    actions   = [
      "s3:GetBucketLocation",
      "s3:GetObject",
      "s3:ListBucket",
      "s3:ListBucketMultipartUploads",
      "s3:ListMultipartUploadParts",
      "s3:AbortMultipartUpload",
      "s3:CreateBucket",
      "s3:PutObject"
    ]
    resources = [
      "arn:aws:s3:::some-bucket-name",
      "arn:aws:s3:::some-bucket-name/*"
    ]
    condition {
      test     = "StringEquals"
      variable = "aws:RequestTag/Tribe"
      values = [
        "someTribe",
      ]
    } 
  }
}
```

Passes:
```
data "aws_iam_policy_document" "wham-dev" {
  statement {
    sid       = "SsoInLinePolicy1"
    effect    = "Deny"
    actions   = [
      "quicksight:*",
      "ds:AuthorizeApplication",
      "ds:UnauthorizeApplication",
      "ds:CheckAlias",
      "ds:CreateAlias",
      "ds:DescribeDirectories",
      "ds:DescribeTrusts",
      "ds:DeleteDirectory",
      "ds:CreateIdentityPoolDirectory",
      "iam:ListAccountAliases"
    ]
    resources = ["*"]
    condition {
      test     = "StringEquals"
      variable = "aws:RequestTag/Tribe"
      values = [
        "someTribe",
      ]
    }
  }
  statement {
    sid       = "SsoInLinePolicy2"
    effect    = "Allow"
    actions   = [
      "s3:GetBucketLocation",
      "s3:GetObject",
      "s3:ListBucket",
      "s3:ListBucketMultipartUploads",
      "s3:ListMultipartUploadParts",
      "s3:AbortMultipartUpload",
      "s3:CreateBucket",
      "s3:PutObject"
    ]
    resources = [
      "arn:aws:s3:::some-bucket-name",
      "arn:aws:s3:::some-bucket-name/*"
    ]
    condition {
      test     = "StringEquals"
      variable = "aws:RequestTag/Tribe"
      values = [
        "someTribe",
      ]
    } 
  }
}
```

**Expected behavior**
I had a look on previous issues similar to this one and my understanding (although could be wrong) is that it should treat the condition block as a restriction regardless of the Effect value.

I have also seen:
https://github.com/salesforce/cloudsplaining/pull/167/files and https://github.com/bridgecrewio/checkov/issues/990
where I believe was fixed.

Is there anything that I am missing?

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.4 
 - Checkov --version
2.0.276

Thank you

**Comment by MooreDerek** [comment]
@hpapadopoulos The issue is with the wildcard in the resources. The code is doing the following:
* Uses methods from the PolicyDocument from `https://github.com/salesforce/cloudsplaining/blob/master/cloudsplaining/scan/policy_document.py`
* That checks that the policy has `permissions_management_without_constraints`
* In turn that checks on the `statement_detail.py` for all "Allow" statements that have `permissions_management_actions_without_constraints`
* which runs the check `has_resource_constraints `
* this has a check `has_resource_wildcard ` and this is catching the "*" in the failing example

The code is determines that due to the wildcard on the resource is not fully constrained and so it is flagged.

So in short the test does take Effect into account :)
